### ISSUE TITLE
'Tensor' object has no attribute dimshuffle

### ISSUE THREAD

**Issue by 1158191255** [jacoxu/ASAM#2]
在extend_layers.py文件的第111行，有一条语句是mask=mask.dimshuffle(0,1,'x')，但是运行的时候却报了题目中所示的错误。求问这是怎么回事呢，是和库的版本有关系么

**Comment by 1158191255** [comment]
这个错误原因是因为backend系统默认使用tensorflow，而tensorflow中是没有dimshuffle的。
解决方法是在主文件夹下的.keras文件夹中(楼主用的是Ubuntu系统)找到keras.json文件，将其中的"backend"属性设置为theano，当然还要保证系统中有这个库。至于.keras文件夹可能需要运行一次程序后才有。改完后再运行程序就可以解决这个问题了

**Comment by 1158191255** [comment]
顺便一提，改完我楼上说的这点之后，再运行程序，它还会报一个错误，说是"An update must have the same type as the orginal shared variable"这个错误是由于更新变量的格式与预设的不一致造成的，结果方法同样是修改keras.json，将其中的第一行"floatx"改为报错信息中update_cal.type=TensorType(此处的格式，scalar)即可，我的位置是float64。改完之后就可以正常训练了
